### PR TITLE
File 'content_sao' splitted into folder 'src/server/object'.

### PR DESCRIPTION
**Pull request by Unarelith**
So as @SmallJoker asked, I'm splitting #7900 into two distinct PRs.

### CODE REVIEW

**Comment by rubenwardy** [comment]
PR lgtm, but I haven't done a diff to check for modifications

To do:

- [ ] Update Android makefile source list
- [ ] Get approval for moving to .hpp

File: src/server/object/TestSAO.hpp [hunk 1]
```diff
@@ -0,0 +1,54 @@
+/*
```
Comment (paramat) [inline src/server/object/TestSAO.hpp]: Filename .hpp -> .h

**Comment by paramat** [comment]
Thanks.
I like what this does, but if it conflicts with any of the blocker PRs i think those have priority for merge https://github.com/minetest/minetest/pulls?q=is%3Aopen+is%3Apr+label%3ABlocker

**Comment by paramat** [comment]
Please can we stay with the project convention of lowercase with underscores for filenames? It's more readable than a solid string of camelcase characters.

**Comment by paramat** [comment]
More thoughts on filenames, see https://github.com/minetest/minetest/pull/7908#issuecomment-442669907 onwards.

**Comment by paramat** [comment]
@nerzhul 's request "header files should be .h instead of .hpp" seems done.

**Comment by paramat** [comment]
I prefer `.h` too:
* Less work.

* `mapgen.cpp` looks visually quite similar to `mapgen.hpp`, only one character differs, so it is visually easier to distinguish files when using `.h`.

**Comment by Unarelith** [comment]
Hey @paramat, I forgot this one. The goal of this PR was also to split a big file into multiple smaller files.

**Comment by nerzhul** [comment]
i missed that PR, but now it's useless, i did the refacto + some code cleanups myself last week

**Pull request closed without merging**